File: reversort/reversort.cpp
/*

Reversort Algorithm

Reversort(L):
	for i:= 1 to length(L)-1
		j = position of min element from i th position to the length of the list (inclusive)
		
		// Reverse teh list from i th position to the j th position (min element)
		Reverse(L[i : j])

*/

#include <bits/stdc++.h>
using namespace std;

#define vi vector<int>
#define vll vector<long long int>

int reversort(vi &A, int N){

	int C = 0;

	for(int i=0; i<N-1; i++){
		int min = *min_element(A.begin()+i, A.end());

		auto it = find(A.begin()+i, A.end(), min);

		int dist = it - (A.begin()+i) + 1;

		reverse(A.begin()+i, it+1);

		/*for(auto a: A){
			cout << a << " ";
		}
		cout << endl;*/

		C = C + dist;
	}

	return C;
}

int main(){

	#ifndef ONLINE_JUDGE
	    // for getting input from input.txt
	    freopen("input.txt", "r", stdin);
	    // for writing output to output.txt
	    freopen("output.txt", "w", stdout);
	#endif
	
	int T;

	cin >> T;

	for(int i=0; i<T; i++){

		int N;
		cin >> N;
		
		vi A;

		for(int j=0; j<N; j++){
			int e;
			cin >> e;

			A.push_back(e);
		}

		cout << "Case #" << i << ": " << reversort(A, N) << endl;
	}

}

File: reversort/reversort.py
"""

Reversort Algorithm

Reversort(L):
	for i:= 1 to length(L)-1
		j = position of min element from i th position to the length of the list (inclusive)
		
		# Reverse teh list from i th position to the j th position (min element)
		Reverse(L[i : j])

"""

import os, sys

if not os.environ.get("ONLINE_JUDGE"):
	sys.stdin = open('input.txt', 'r')
	sys.stdout = open('output.txt', 'w')


def func(arr, l):
	C = 0

	for i in range(0, l-1):
		m = min(arr[i:])

		ind = arr.index(m)
		dis = ind - i + 1

		arr = arr[:i] + arr[i:ind+1][::-1] + arr[ind+1:]

		C += dis

	return C


N = int(input())

for case in range(N):
	l = int(input())
	arr = list(map(int, input().split(' ')))

	print("Case #{}: {}".format(case, func(arr, l)))